xmtp_mls/src/groups/mod.rs

```diff
@@ -518,10 +552,15 @@ where
             let (payload, post_commit_data) = result.expect("result already checked");
             let payload_slice = payload.as_slice();
 
-            self.client
-                .api_client
-                .publish_to_group(vec![payload_slice])
-                .await?;
+            retry_async!(
```

Comment:
WDYT about moving the retry logic into the `ApiClientWrapper` so that all callers don't have to think about this

Reply:
That makes sense,

Actually, should we be retrying api client errors? There's an issue on the board talking about idempotency in the API

Reply:
I'm fine with building it in for now and assume the idempotency is going to be there. If it starts making test cases unreliable we can always disable or set the number of retries to 0

Reply:
Also, it's only publish to group and publish welcomes that are not idempotent. Registering identities is, and all the query APIs are obviously safe.